uberpoet/cpprojectgen.py

```diff
@@ -75,26 +79,51 @@ def example_command(self):
 
     # Generation Functions
 
-    def gen_app(self, app_node, node_list, target_loc, loc_json_file_path):
+    def gen_app(self, app_node, node_list, target_swift_loc, target_objc_loc, loc_json_file_path):
         library_node_list = [n for n in node_list if n.node_type == ModuleNode.LIBRARY]
 
         if loc_json_file_path:
             loc_reader = locreader.LocFileReader()
             loc_reader.read_loc_file(loc_json_file_path)
             loc = loc_reader.loc_for_module(n.name)
-            module_index = {n.name: {"files": self.gen_lib_module(n, loc), "loc": loc} for n in library_node_list}
+            module_index = {
+                n.name: {
+                    "files": self.gen_lib_module(n, loc, Language.SWIFT),
+                    "loc": loc,
+                    "language": Language.SWIFT
+                } for n in library_node_list
+            }
         else:
             total_code_units = 0
             for l in library_node_list:
                 total_code_units += l.code_units
 
-            loc_per_unit = target_loc / total_code_units
-            module_index = {
-                n.name: {
-                    "files": self.gen_lib_module(n, loc_per_unit),
-                    "loc": loc_per_unit
-                } for n in library_node_list
-            }
+            total_loc = target_swift_loc + target_objc_loc
+
+            swift_module_count_percentage = round(float(target_swift_loc) / total_loc, 2)
+            objc_module_count_percentage = round(float(target_objc_loc) / total_loc, 2)
+
+            split_library_node_list = list(
```

Comment:
this splits the generated module list across the percentage of LOC that needs to be for ObjC and Swift depending on the parameters passed.

Reply:
ha, yes makes sense